x-pack/legacy/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -350,6 +370,43 @@ export class VectorLayer extends AbstractLayer {
     };
   }
 
+  async _performInnerJoins(sourceResult, joinStates, updateSourceData) {
+
+    let shouldUpdate = false;
+    for (let i = 0; i < sourceResult.featureCollection.features.length; i++) {
+      const feature = sourceResult.featureCollection.features[i];
+      const oldVisbility = feature.properties[FEATURE_VISIBLE_PROPERTY_NAME];
+      feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = true;//new visibility will be determined based on join-results
+      for (let j = 0; j < joinStates.length; j++) {
+        const joinState = joinStates[j];
+        const leftInnerJoin = joinState.join;
+        const rightMetricFields = leftInnerJoin.getRightMetricFields();
+
+        let canJoin;
+        if (sourceResult.refreshed || joinState.dataHasChanged) {
```

Comment:
pull `canJoin = leftInnerJoin.joinPropertiesToFeature(feature, joinState.propertiesMap, rightMetricFields);` out of if/else. Then just have if statement that only sets shouldUpdate

Reply:
pulling this out in separate guard statement on top of function to make these nested loop less cluttered.